cadl-extension/src/code-model-builder.ts

```diff
@@ -1798,37 +1779,50 @@ export class CodeModelBuilder {
   }
 
   private _anySchema?: AnySchema;
-  public get anySchema(): AnySchema {
+  get anySchema(): AnySchema {
     return this._anySchema ?? (this._anySchema = this.codeModel.schemas.add(new AnySchema("Anything")));
   }
 
+  private createApiVersionParameter(serializedName: string, parameterLocation: ParameterLocation): Parameter {
+    return new Parameter(
+      serializedName,
+      "Version parameter",
+      this.codeModel.schemas.add(
+        new ConstantSchema(serializedName, "API Version", {
+          valueType: this.stringSchema,
+          value: new ConstantValue(""),
+        }),
+      ),
+      {
+        implementation: ImplementationLocation.Client,
+        origin: "modelerfour:synthesized/api-version",
+        required: true,
+        protocol: {
+          http: new HttpParameter(parameterLocation),
+        },
+        language: {
+          default: {
+            serializedName: serializedName,
+          },
+        },
+      },
+    );
+  }
+
   private _apiVersionParameter?: Parameter;
   get apiVersionParameter(): Parameter {
     return (
       this._apiVersionParameter ||
-      (this._apiVersionParameter = new Parameter(
-        "api-version",
-        "Version parameter",
-        this.codeModel.schemas.add(
-          new ConstantSchema("accept", "api-version", {
-            valueType: this.stringSchema,
-            value: new ConstantValue(""),
-          }),
-        ),
-        {
-          implementation: ImplementationLocation.Client,
-          origin: "modelerfour:synthesized/api-version",
-          required: true,
-          protocol: {
-            http: new HttpParameter(ParameterLocation.Query),
-          },
-          language: {
-            default: {
-              serializedName: "api-version",
-            },
-          },
-        },
-      ))
+      (this._apiVersionParameter = this.createApiVersionParameter("api-version", ParameterLocation.Query))
+    );
+  }
+
+  private _apiVersionParameterInPath?: Parameter;
+  get apiVersionParameterInPath(): Parameter {
+    return (
+      this._apiVersionParameterInPath ||
+      // TODO: hardcode as "apiVersion", as it is what we get from compiler
+      (this._apiVersionParameterInPath = this.createApiVersionParameter("apiVersion", ParameterLocation.Path))
```

Comment:
hardcoded as "apiVersion" for now.

If emitter logic changes, code will fail the test and we shall change this.